aws/resource_aws_servicecatalog_provisioning_artifact.go

```diff
@@ -0,0 +1,297 @@
+	d.Set("name", pad.Name)
+	d.Set("product_id", productID)
+	d.Set("type", pad.Type)
+
+	return nil
+}
+
+func resourceAwsServiceCatalogProvisioningArtifactUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+
+	if d.HasChanges("accept_language", "active", "description", "guidance", "name", "product_id") {
+		artifactID, productID, err := tfservicecatalog.ProvisioningArtifactParseID(d.Id())
+
+		if err != nil {
+			return fmt.Errorf("error parsing Service Catalog Provisioning Artifact ID (%s): %w", d.Id(), err)
+		}
+
+		input := &servicecatalog.UpdateProvisioningArtifactInput{
+			ProductId:              aws.String(productID),
+			ProvisioningArtifactId: aws.String(artifactID),
+			Active:                 aws.Bool(d.Get("active").(bool)),
+		}
+
+		if v, ok := d.GetOk("accept_language"); ok {
+			input.AcceptLanguage = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("description"); ok {
+			input.Description = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("guidance"); ok {
+			input.Guidance = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("name"); ok {
+			input.Name = aws.String(v.(string))
+		}
+
+		err = resource.Retry(iamwaiter.PropagationTimeout, func() *resource.RetryError {
+			_, err := conn.UpdateProvisioningArtifact(input)
+
+			if tfawserr.ErrMessageContains(err, servicecatalog.ErrCodeInvalidParametersException, "profile does not exist") {
+				return resource.RetryableError(err)
+			}
+
+			if err != nil {
+				return resource.NonRetryableError(err)
+			}
+
+			return nil
+		})
+
+		if tfresource.TimedOut(err) {
+			_, err = conn.UpdateProvisioningArtifact(input)
+		}
+
+		if err != nil {
+			return fmt.Errorf("error updating Service Catalog Provisioning Artifact (%s): %w", d.Id(), err)
+		}
+	}
+
+	return resourceAwsServiceCatalogProvisioningArtifactRead(d, meta)
+}
+
+func resourceAwsServiceCatalogProvisioningArtifactDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+
+	artifactID, productID, err := tfservicecatalog.ProvisioningArtifactParseID(d.Id())
+
+	if err != nil {
+		return fmt.Errorf("error parsing Service Catalog Provisioning Artifact ID (%s): %w", d.Id(), err)
+	}
+
+	input := &servicecatalog.DeleteProvisioningArtifactInput{
+		ProductId:              aws.String(productID),
+		ProvisioningArtifactId: aws.String(artifactID),
+	}
+
+	if v, ok := d.GetOk("accept_language"); ok {
+		input.AcceptLanguage = aws.String(v.(string))
+	}
+
+	_, err = conn.DeleteProvisioningArtifact(input)
+
+	if tfawserr.ErrCodeEquals(err, servicecatalog.ErrCodeResourceNotFoundException) {
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error deleting Service Catalog Provisioning Artifact (%s): %w", d.Id(), err)
+	}
+
+	if err := waiter.ProvisioningArtifactDeleted(conn, artifactID, productID); err != nil {
+		return fmt.Errorf("error waiting for Service Catalog Provisioning Artifact (%s) to be deleted: %w", d.Id(), err)
```

Comment:
similar comment as above about redundant error log message; here this would inherit the `error waiting for state of provisioning artifact` error message as well

Reply:
hmm maybe could update `error waiting for state of provisioning artifact` to `error waiting for state of provisioning artifact to be NOT_FOUND`?

Reply:
yeah, Go can be a little excessive with error wrapped with error wrapped with... fixed.